### PR TITLE
Add corrupted files in bad_data

### PR DESCRIPTION
**Pull request by rouault**
Those 2 files triggered libparquet c++ issues https://github.com/apache/arrow/issues/41317 and https://github.com/apache/arrow/issues/41321 . They have been generated through a local run of oss-fuzz on synthetic test data of the GDAL regression test suite, and can be licensed under Apache-2

### CODE REVIEW

**Review by mapleFU — APPROVED** [review]

**Review by alamb — APPROVED** [review]
Makes sense to me. Thank you @rouault  and @mapleFU 

I think we may need a parquet reviewer to approve / merge this PR as well

**Review by wgtmac — APPROVED** [review]

File: bad_data/README.md [hunk 1]
```diff
@@ -22,3 +22,7 @@ These are files used for reproducing various bugs that have been reported.
 
 * PARQUET-1481.parquet: tests a case where a schema Thrift value has been
   corrupted
+* arrow_issue_41321.parquet: test case of https://github.com/apache/arrow/issues/41321
```
Comment (pitrou) [inline bad_data/README.md:25]: Maybe try to unify the file naming in this directory? We already have `PARQUET-1481.parquet` (a JIRA reference) so perhaps something like `ARROW-GH-41321.parquet`? (related: https://github.com/apache/parquet-testing/pull/57)

Comment (mapleFU) [reply bad_data/README.md:25]: Let me edit it 🤔

Comment (mapleFU) [reply bad_data/README.md]: done

**Comment by mapleFU** [comment]
I would merge this in 1 day if no negative comment. 7.1k is a bit large here but it's not too large since generate a file like this is also hard. I

**Comment by mapleFU** [comment]
Merged! Thanks all!

**Pull request merged**